docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,237 @@
+  - A large number of companies use CSS-in-JS in production, including Reddit, Patreon, Target, Atlassian, Vogue, GitHub, Coinbase, and many more.
+- Cons
+  - Putting a CSS-in-JS library into use will add an extra layer of dependencies to our front-end stack, although that could be mitigated by simply using CSS-in-JS files and not any library if we're using React.
+  - CSS-in-JS libraries generate selectors automatically which might worsen code readability, and can be difficult for new contributors.
+
+### Option D - Use a CSS pre-processor like SASS
+
+- Description
+  - [SASS](https://sass-lang.com/documentation) (which stands for 'Syntactically awesome style sheets) is an extension of CSS that enables making use of variables, nested rules, inline imports and other interesting functionalities that can make writing CSS much more powerful. in a stylesheet. It comes with 2 kinds of syntax: SCSS (Sassy CSS) and Indented (simply called 'Sass'). SCSS files use .scss extension and are fully compliant with CSS syntax, whereas the SASS files are not and hence, we will be discussing only about using .scss files as our styling solution in this section.
+- Pros
+  - Sass offers little-to-no learning curve especially if we're using SCSS syntax since it's nothing but CSS.
+  - It has large community and well documented.
+  - SASS offers modularization. With the help of `@import` directive, we can import partials (small SASS files) into other SASS files.
+  - It is compatible with all versions of CSS so any available CSS libraries can be employed with SASS.
+- Cons
+  - Using SASS will add an extra step of compilation which can potentially slow down development.
+  - Source files may be small but the generated CSS could be huge.
+
+## Pros and Cons of the Email Previewing Options
+
+It’s notoriously difficult to preview what different email states look like in FxA for Engineering, Design, and Product. Currently, we're manually previewing the templates by running `yarn write-emails` in the `fxa-auth-server` root directory which doesn't account for states unless we manually change variables and run the command again. Furthermore, even the Maildev client requires us to manually trigger the email when running FxA locally so we need a robust solution to account for these challenges faced by the team.
+
+### Option A - Continue to use the `write-emails` command and maildev client
+
+- Description
+  - Continue exclusively using the `yarn write-emails` command with [maildev](https://www.npmjs.com/package/maildev) to preview emails.
+- ## Pros
+- Cons
+  - Previewing emails with maildev will require us to manually trigger emails while FxA is running locally.
+  - Since Maildev doesn't account for viewing various states of the emails simultaneously, it could be a bit complex to change variables and triggering the mail again and again.
```

Comment:
I noted this above, but MailDev is helpful to test mail delivery manually and can also be helpful when signing up for an account locally and retrieving your account verification code. So it has applications beyond previewing templates.

Reply:
Yeah, it's probably worth noting somewhere here that even if we don't go with option A, we plan to continue supporting use of MailDev.

Reply:
Yea, I think we can probably just remove references to maildev since it is used to recieve emails from users while they are going through app flows.